frontend/public/components/utils/horizontal-nav.tsx

```diff
@@ -116,7 +116,7 @@ export const navFactory: NavFactory = {
   }),
 };
 
-export const NavBar: React.SFC<NavBarProps> = ({pages, basePath, hideDivider}) => {
+export const NavBar = withRouter<NavBarProps>(({pages, basePath, hideDivider}) => {
```

Comment:
This is actually a better solution to ensure `NavBar` re-renders when the route changes. Before it was depending on `HorizontalNav` to be re-rendered, but using `withRouter` ensures we capture route changes.